sdk/src/Core/Amazon.Runtime/Credentials/Internal/_bcl45+netstandard/SsoTokenUtils.cs

```diff
@@ -92,6 +92,23 @@ public static string AsJson(this SsoToken token)
             return ToJson(token);
         }
 
+        /// <summary>
+        /// This determines whether the SSO token's registrationExpiresAt field is 5 minutes within expiration
+        /// </summary>
+        /// <param name="token"></param>
+        /// <returns></returns>
+        public static bool RegisteredClientExpired(this SsoToken token)
+        {
+            DateTime dateTime;
+            if(ConvertStringToDateTime(token.RegistrationExpiresAt,out dateTime))
+            {
+#pragma warning disable CS0618 // Type or member is obsolete
+                return AWSSDKUtils.CorrectedUtcNow >= dateTime.AddMinutes(-5);
```

Comment:
Just curious, how'd you arrive at 5 minutes?

Reply:
My thought process was similar to how we refresh credentials if `PreEmptExpiryTime` is within 15 minutes of expiration. Also in SsoTokenUtils, it uses 6 minutes to determine if the token should refresh, so I just did something similar.